pkg/cluster/failurediagnostics/virtualmachines.go

```diff
@@ -0,0 +1,75 @@
+package failurediagnostics
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"bufio"
+	"context"
+	"encoding/base64"
+	"fmt"
+
+	"github.com/Azure/ARO-RP/pkg/util/stringutils"
+)
+
+func (m *manager) LogAzureInformation(ctx context.Context) (interface{}, error) {
+	items := make([]interface{}, 0)
+
+	if m.virtualMachines == nil {
+		items = append(items, "vmclient missing")
+		return items, nil
+	}
+
+	resourceGroupName := stringutils.LastTokenByte(m.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID, '/')
+	vms, err := m.virtualMachines.List(ctx, resourceGroupName)
+	if err != nil {
+		items = append(items, fmt.Sprintf("vm listing error: %s", err))
+		return items, nil
+	}
+
+	if len(vms) == 0 {
+		items = append(items, "no VMs found")
+		return items, nil
+	}
+
+	vmNames := make([]string, 0)
+	for _, v := range vms {
+		j, err := v.MarshalJSON()
+		if err != nil {
+			items = append(items, fmt.Sprintf("vm marshalling error: %s", err))
+		} else {
+			vmName := "<unknown>"
+			if v.Name != nil {
+				vmName = *v.Name
+				vmNames = append(vmNames, vmName)
+			}
+			items = append(items, fmt.Sprintf("vm %s: %s", vmName, string(j)))
+		}
+	}
+
+	// Fetch boot diagnostics URIs for the VMs
+	for _, vmName := range vmNames {
+		func() {
```

Comment:
It's curious to me that this is executed in a anon function, is this just so we can do the defer easily? I think there's some funky issues that we'll experience with Go 1.22 because of this - and also I'm not sure if we've turned on the new loop compilation so that you don't have to do vmName := vmName at the top of the loop to ensure it doesn't try to get the same vmname everytime.